shared/utils/combine-ovals.py

```diff
@@ -278,18 +278,21 @@ def checks(product, oval_version, oval_dirs):
             # sort the files to make output deterministic
             for filename in sorted(os.listdir(oval_dir)):
                 if filename.endswith(".xml"):
-                    with codecs.open(os.path.join(oval_dir, filename), "r",
-                                     encoding="utf-8") as xml_file:
-                        xml_content = xml_file.read()
-                        if not check_is_applicable_for_product(xml_content, product):
-                            continue
-                        if check_is_loaded(already_loaded, filename, oval_version):
-                            continue
-                        if not check_oval_version_from_oval(xml_content, oval_version):
-                            continue
-                        body.append(xml_content)
-                        included_checks_count += 1
-                        already_loaded[filename] = oval_version
+                    xml_content = ssgcommon.process_file_with_jinja(
+                        os.path.join(oval_dir, filename), product_yaml
+                    )
+                    if not check_is_applicable_for_product(
+                        xml_content,
+                        ssgcommon.required_yaml_key(product_yaml, "product")
+                    ):
```

Comment:
Do you see jinja being used to determine applicability? Otherwise we can check for applicability before processing.

Reply:
It's not used for applicability right now but I am hoping it will replace the applicability checking in the future. We have to process first because the other order breaks parsing error functionality. It's doable but not without loading the file twice.

Reply:
Thanks for clarification.